middle_end/flambda2/to_cmm/to_cmm_primitive.ml

```diff
@@ -123,10 +123,7 @@ let block_load ~dbg (kind : P.Block_access_kind.t) (mutability : Mutability.t)
     match field_kind with
     | Tagged_immediate ->
       C.get_field_computed Immediate mutability ~block ~index dbg
-    | Naked_float ->
-      (* CR layouts v5.1: We should use the mutability here to generate better
-         code if the load is immutable. *)
-      C.unboxed_float_array_ref block index dbg
+    | Naked_float -> C.unboxed_immutable_float_array_ref block index dbg
```

Comment:
Shouldn't this use `mutability` to detect whether the load is mutable or not? It's possible for a naked float field in a mixed block to be mutable.